### PR TITLE
CRM-18367 Remove long-broken contribution logging reports

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Remove report templates that have been broken since at least 4.4

Before
----------------------------------------
Contribution Logging reports ship with core but don't work

After
----------------------------------------
Contribution Logging reports no longer ship with core

Technical Details
----------------------------------------

Comments
----------------------------------------
Note that there is some appetite to get this working per JIRA discussion. I have added a limited version to the extended reports extension and it may be we re-add to core but I don't think this code is worth resuscitating as the basis of a future effort.

pin @davejenx @lcdservices

---

 * [CRM-18367: Fatal error on Contribution logging summary report \(possibly remove - see comment\)](https://issues.civicrm.org/jira/browse/CRM-18367)

### CODE REVIEW

**Comment by seamuslee001** [comment]
should we be checking if there are any menu items built off these reports?

**Comment by seamuslee001** [comment]
also what about records in `civicrm_report_instance` ?

**Comment by eileenmcnaughton** [comment]
Good thinking. 

It looks like civicrm_menu does not have entries & civicrm_navigation does not have entries that are independent of the references from civicrm_report_instance so I think the changes I just mad should cover it

File: CRM/Upgrade/Incremental/php/FourSeven.php [hunk 1]
```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
...
 
   /**
+   * Remove the contribution logging reports which have been broken for a very long time.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function removeContributionLoggingReports(CRM_Queue_TaskContext $ctx) {
+    if (class_exists('CRM_Report_Form_Contribute_LoggingDetail') || class_exists('CRM_Report_Form_Contribute_LoggingSummary')) {
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: note you could argue that if only one has been deleted we should still process the other. I leaned towards early back out if any evidence of custom stuff

Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: +1 on this

File: CRM/Upgrade/Incremental/php/FourSeven.php [hunk 2]
```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
...
+      // Perhaps the site has overridden these classes. The core ones are broken but they
+      // may have functional ones.
+      return TRUE;
+    }
+    $options = civicrm_api3('OptionValue', 'get', array('option_group_id' => 'report_template', 'options' => array('limit' => 0)));
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: isn't this getting all reports? shouldn't we limit them to just the 2 that we care about?

File: CRM/Upgrade/Incremental/php/FourSeven.php [hunk 3]
```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
...
+    foreach ($options['values'] as $option) {
+      $instances = civicrm_api3('ReportInstance', 'get', array('report_id' => $option['value']));
+      if ($instances['count']) {
+        foreach ($instances['values'] as $instance) {
+          if ($instance['navigation_id']) {
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FourSeven.php]: shouldn't this be nested under the If in new Line 1259?

File: CRM/Upgrade/Incremental/php/FourSeven.php (hunk 2 above)
Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: yeah, getting all & iterating to filter. I guess I could use IN - but I tested this one!

File: CRM/Upgrade/Incremental/php/FourSeven.php (hunk 3 above)
Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: hot diggity dog! you are right!

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: fix pushed

Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: probably everything from L1250 to L1258 needs to be under the if

File: CRM/Upgrade/Incremental/php/FourSeven.php (hunk 2 above)
Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FourSeven.php]: Makes sense i guess

**Comment by seamuslee001** [comment]
@eileenmcnaughton should this get a postupgrade message attached?

**Comment by eileenmcnaughton** [comment]
I did think about the message - but it feels kinda confusing to say "something that has been broken for 3 years is now not findable - but obviously you never found it anyway because you didn't know it was broken

**Comment by seamuslee001** [comment]
@eileenmcnaughton this needs rebasing, I haven't tested this yet bu i think it makes sense to me pinging @monishdeb on this one

**Comment by eileenmcnaughton** [comment]
I just rebased it - although we have until Wed before it needs tweaking for the next release

**Review by monishdeb — APPROVED** [review]
I tested the patch on my local, works fine. From code pov it makes sense.

**Comment by monishdeb** [comment]
@eileenmcnaughton are you going to bring any change? or can I merge it?

**Comment by eileenmcnaughton** [comment]
@monishdeb from my point of view it is good to merge

**Comment by monishdeb** [comment]
Done

**Comment by monishdeb** [comment]
@eileenmcnaughton can you review https://github.com/civicrm/civicrm-core/pull/11059 in exchange? :)

**Pull request merged**